cardinal/router/router_test.go

```diff
@@ -203,6 +205,42 @@ func TestRouter_SendMessage_TxFailed(t *testing.T) {
 	assert.Equal(t, res.Code, CodeTxFailed)
 }
 
+var _ shard.TransactionHandlerClient = &fakeTxHandler{}
+
+type fakeTxHandler struct {
+	req *shard.RegisterGameShardRequest
+}
+
+func (f *fakeTxHandler) RegisterGameShard(
+	_ context.Context,
+	in *shard.RegisterGameShardRequest,
+	_ ...grpc.CallOption,
+) (*shard.RegisterGameShardResponse, error) {
+	f.req = in
+	return &shard.RegisterGameShardResponse{}, nil
+}
+
+func (f *fakeTxHandler) Submit(
+	_ context.Context,
+	_ *shard.SubmitTransactionsRequest,
+	_ ...grpc.CallOption,
+) (*shard.SubmitTransactionsResponse, error) {
+	panic("implement me")
```

Comment:
tbh, I don't mind stuff like this. But typically If I did it someone would block this in my PR, hard. Add a todo here at least.

Reply:
yeah, its just there so it satisfies the interface so i can pass the object in. i changed the panic message: [change panic message in mock](https://github.com/Argus-Labs/world-engine/pull/632/commits/69c95a0b027c3510ba4f6338ef83f304245b0977)